### ISSUE TITLE
FFT-based Acceleration

### ISSUE THREAD

**Issue by nobodyplayer** [LMissher/TFMAE#4]
Hello, I would like to ask which code corresponds to the FFT acceleration mentioned in your paper? Seems to use only convolution to accelerate it?

**Comment by frakeyc** [comment]
Hello, only convolution is used because convolution in the time domain is equivalent to multiplication in the frequency domain, which is faster than using FFT and iFFT operation in PyTorch.

**Comment by LMissher** [comment]
Hello, only convolution is used because convolution in the time domain is equivalent to multiplication in the frequency domain, which is faster than using FFT and iFFT operation in PyTorch.

**Comment by nobodyplayer** [comment]
Thank you for your reply! I also want to ask one more question, the paper describes the two views of the time domain and the frequency domain to counter training, but the code shows that you only use the frequency domain to perform frequency mask and then turn back to the time domain for encoding, my understanding is that this is equivalent to the comparison of two mask methods, rather than the representation of the time domain and the frequency domain.

**Comment by nobodyplayer** [comment]
By the way, why compare att and rec? And it seems that the rec comparison is important to the results

**Comment by LMissher** [comment]
We propose two novel masking strategies to eliminate anomalies in the temporal-frequency domains. The "frequency and temporal **masking-based representations**" but not the "frequency and temporal representations" are then derived by networks. Therefore, I think representations are compared but not the masked values. Besides, rec denotes final representations and naturally should be used to compare.

**Comment by nobodyplayer** [comment]
I completely understand, thank you for your answer! Your work is excellent




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年12月10日(星期二) 晚上7:52
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [LMissher/TFMAE] FFT-based Acceleration (Issue #4)














 
我们提出了两种新颖的掩蔽策略来消除时频域中的异常。然后，“基于频率和时间掩码的表示”而不是“频率和时间表示”由网络派生。因此，我认为会比较表示，但不会比较掩码值。此外，rec 表示最终表示，自然应该用于比较。
 
—
直接回复此电子邮件，在 GitHub 上查看，或取消订阅。
您收到此消息是因为您编写了该线程。Message ID: ***@***.***&gt;

**Comment by LMissher** [comment]
Thank you for your attention and welcome to discuss.
